docs/installation/docker.md

```diff
@@ -0,0 +1,121 @@
+# Docker Compose (ISLE)
+
+The ISLE Islandora Enterprise 8 Prototype ([Islandora-Devops/isle-dc](https://github.com/Islandora-Devops/isle-dc)) is a Docker project to manage Islandora 8 infrastructure under Docker using Docker Compose. These instructions cover the basics of using Docker locally to create a development Islandora sandbox. More detailed configuration options are described in the project's [README](https://github.com/Islandora-Devops/isle-dc).
+
+## Why use Docker?
+
+Docker is a way to separate out the "state" of your site (i.e. all the content, files, and configurations that you've entered) from the underlying software that runs it (e.g. webserver, database engine, etc). This allows for easier upgrades, faster development, and more flexible deployment. The ISLE project, run by the Islandora Collaboration Group (ICG) was the driving force behind dockerizing Islandora.
+
+## Requirements
+
+* Docker (version 19.x+)
+* (Mac OS) XCode Command-Line Tools
+* Perl (if using `make dev`)
```

Comment:
It seems from the README that there are many more requirements. Shoot, I already had them so didn't run into this. Are they all required in all cases? Or can one do the simple `make` without all of these installed on the host?

Reply:
Yes - these should only be required for 'make dev' and i'll update the docs.

Reply:
I just tried to run this on my Windows 10 machine and I don't have make installed. I tried "npm install make -g" and it runs but immediately gives an error so I'm not sure I got the correct thing. Output below

`$ make dev
make i info Invoking dev target
'$' is not recognized as an internal or external command,
operable program or batch file.
make × ERR  $(MAKE) download-default-certs
$(MAKE) create-codebase-from-demo
if grep -q DRUPAL_DEFAULT_CONFIGDIR docker-compose.env.yml; then \
perl -i -pe's/DRUPAL_DEFAULT_CONFIGDIR:.*/DRUPAL_DEFAULT_CONFIGDIR: \/var\/www\/drupal\/config\/sync/g'  docker-compose.env.yml; \
else \
perl -i -pe's/DRUPAL_DEFAULT_SALT/DRUPAL_DEFAULT_CONFIGDIR: \/var\/www\/drupal\/config\/sync\n\ \ \ \ \ \ DRUPAL_DEFAULT_SALT/g' docker-compose.env.yml; \
fi
if grep -q  DRUPAL_DEFAULT_INSTALL_EXISTING_CONFIG docker-compose.env.yml; then \
perl -i -pe's/DRUPAL_DEFAULT_INSTALL_EXISTING_CONFIG:.*/DRUPAL_DEFAULT_INSTALL_EXISTING_CONFIG: "true"/g' docker-compose.env.yml; \
else \
perl -i -pe's/DRUPAL_DEFAULT_SALT/DRUPAL_DEFAULT_INSTALL_EXISTING_CONFIG: "true"\n\ \ \ \ \ \ DRUPAL_DEFAULT_SALT/g' docker-compose.env.yml; \
fi
if grep -q  DRUPAL_DEFAULT_PROFILE docker-compose.env.yml; then \
perl -i -pe's/DRUPAL_DEFAULT_PROFILE:.*/DRUPAL_DEFAULT_PROFILE: minimal/g' docker-compose.env.yml; \
else \
perl -i -pe's/DRUPAL_DEFAULT_SALT/DRUPAL_DEFAULT_PROFILE: minimal\n\ \ \ \ \ \ DRUPAL_DEFAULT_SALT/g' docker-compose.env.yml; \
fi
$(MAKE) -B docker-compose.yml ENVIRONMENT=local
docker-compose up -d
$(MAKE) remove_standard_profile_references_from_config
$(MAKE) install ENVIRONMENT=local
$(MAKE) hydrate ENVIRONMENT=local
make × ERR  Recipe exited with code %d
(node:20452) UnhandledPromiseRejectionWarning: Error: Recipe exited with code %d
    at ChildProcess.<anonymous> (C:\Users\catspaw\AppData\Roaming\npm\node_modules\make\src\cli.js:143:28)
    at ChildProcess.emit (events.js:198:13)
    at maybeClose (internal/child_process.js:982:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
(node:20452) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:20452) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`

Reply:
Running "make" by itself just results in a help page because it says it needs a target.

Reply:
all right! Thanks @dflitner

Reply:
Where are you running make @dflitner ?  A Windows command terminal, power shell, git bash shell, wsl or wsl2 shell?  Unfortunately, there are environments one can be in on Windows, and it is not clear which one you are using.

Reply:
As I mentioned on Slack, I used the Windows Subsystem for Linux (WSL2) with Docker Desktop for Windows. On a freshly installed Debian, I had to install make and curl to spin up ISLE:
`sudo apt-get install make`
`sudo apt-get install curl`

There was also a small hiccup around user permissions and a config file for docker. I'll try to verify if that is an issue under Ubuntu-on-WSL too, so we can document it.